File: rb/features.rb
# NobelPrize SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module NobelPrizeFeatures
  def self.make_feature(name)
    case name
    when "base"
      NobelPrizeBaseFeature.new
    when "ratelimit"
      NobelPrizeRatelimitFeature.new
    when "retry"
      NobelPrizeRetryFeature.new
    when "test"
      NobelPrizeTestFeature.new
    when "timeout"
      NobelPrizeTimeoutFeature.new
    else
      NobelPrizeBaseFeature.new
    end
  end
end

File: py/features.py
# NobelPrize SDK feature factory

from feature.base_feature import NobelPrizeBaseFeature
from feature.test_feature import NobelPrizeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NobelPrizeBaseFeature(),
        "test": lambda: NobelPrizeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
